lib/jekyll/filters.rb

```diff
@@ -1,9 +1,5 @@
 # frozen_string_literal: true
 
-require "addressable/uri"
-require "json"
-require "liquid"
```

Comment:
I can see how removing `require` from a frequently called method could lead to performance improvements, but I can’t see how moving these static requires to a different file could result in any meaningful difference (even considering deduplication). I worry that it obfuscates _why_ we are requiring some things if the `require` is no where near where we are using it.

Can you explain a bit more of the motivation behind _this_ change?

Reply:
Understandably, `Jekyll::Filters` is tightly coupled with `lib/jekyll.rb` &mdash; one cannot simply *`require 'jekyll/filters'`* without `require 'jekyll'` first. i.e. `lib/jekyll.rb` has to be loaded first.

Loading `lib/jekyll.rb` will *require* among others:
  - **`liquid`** (already does so in `master`),
  - **`addressable/uri`** (a change made in this PR itself &mdash; this module is necessary in both `Jekyll::Filters` and `Jekyll::URL`) and
  - **`json`** (another change made in this PR &mdash; it is necessary in `Jekyll::Filters`, `serve/live_reload_reactor.rb`, and in `jekyll/drops`)

So, as part of removing duplicate `require` calls in this PR, I removed these static requires.